Move some logs to `DEBUG`

luckysori opened the pull request.

luckysori committed:
Demote several logs in maker_incoming_connections

I think the `INFO` log level should give us information on the
behaviour of our application at a high level. Reporting the state of
individual connections is too noisy and I think it belongs to `DEBUG`.

luckysori committed:
Demote target-reached monitoring log

We already learn about transactions reaching particular milestones
when a corresponding event is appended to the database. Furthermore,
this log can be triggered more than once per unique transaction,
flooding the logs with irrelevant information.

da-kami commented on daemon/src/maker_inc_connections.rs:
```diff
@@ -605,7 +605,7 @@ async fn upgrade(
     let daemon_version = daemon_version.unwrap_or_else(|| String::from("<= 0.4.7"));
 
-    tracing::info!(%taker_id, %taker_address, %negotiated_wire_version, %daemon_version, "Connection upgrade successful");
+    tracing::debug!(%taker_id, %taker_address, %negotiated_wire_version, %daemon_version, "Connection upgrade successful");
```
Could we include `%taker_address` in the above `Connection is ready` message and downgrade this one to `trace`? I feel having both is somewhat redundant.

luckysori replied:
I'm okay with that. See https://github.com/itchysats/itchysats/pull/1874/commits/9ddaf8281d9e6893dcebdd3ca124f319151bd740.

luckysori committed:
Demote connection-upgrade-success log to trace

We also move the `taker_address` to the "Connection is ready" log, to
still provide that context at a higher level.

bonomat reviewed (approved):
LGTM
